### ISSUE TITLE
[Feature] Recording videos showing the actual input for demo purposes [sf#257]

### ISSUE THREAD

**Issue by openMSX-import** [openMSX/openMSX#281]
**Reported by manuelbi on 2014-01-08 21:40 UTC**
flyguille posted in the openMSX forum:
"when I was recording in video the demostration of some soft for youtube. The emulator records what the MSX screen shows, but sometimes for the demostration I wanted to show in realtime on the video what keys I pressed for the MSX react that way!.

Now in wich part of the MSX screen to overlay or in wich way to show the multiple inputs at the same time or when is a burst, how long, it they will move out for dissapearing, that is another thing.

It must be switache ON & OFF easy as using a NON-MSX keys for that because when recording one wants to show what is pressed only on specific things and not all the video. "

So, in summary: for demo videos it could be useful to show on screen which keys/inputs are pressed, to explain how to operate a certain MSX software. And for YouTube purposes, this would then have to be recordable of course... (which is actually a separate feature request: record not only the MSX pixels, but also OSD pixels...)

**Comment by openMSX-import** [comment]
**Commented by flyguille on 2014-01-09 11:53 UTC**
the idea about showing what keys I presses, is more, like showing individual keys graphically I pressed that can appear in the middle of the video, and moving to the left over time for dissapearing from the video, something with a sort of semi-transparent box as base as for showing that it is not part of the msx screen, but the key that I pressed. the graphic can be an abstract icon of the key, but if you go BIG can be an actual photo of the plastic key pressed, and going more BIG, can be a photo of an actual plastic key of the same MACHINE that is emulating.

But who will photograph each individual plastic key by separate on a black background for easy clipping in the edition, then editing it in a reasonable image format?. Overall if the aiming is to show accurately for each MSX machine, asking to who has the specific MSX model to unmount the keys one by one is a monstruos project!!!!.

If there is two or more keys pressed, to show two keys photos or icon graphics with a PLUS sign in the middle plus using the same semi-transparent box as background.

**Comment by MBilderbeek** [comment]
Can be done with the Virtual Keyboard and making a screen recording....
